collector/node_collector.go

```diff
@@ -52,58 +60,59 @@ type metricsMeta struct {
 // NewAgent returns a new instance of a DC/OS agent poller based on the provided
 // configuration and the result of the provided ipCommand script for detecting
 // the agent's IP address.
-func NewAgent(ipCommand string, port int, pollPeriod time.Duration, metricsChan chan<- producers.MetricsMessage) (Agent, error) {
-	a := Agent{}
+func NewDCOSHost(dcosRole string, ipCommand string, port int, pollPeriod time.Duration, metricsChan chan<- producers.MetricsMessage) (DCOSHost, error) {
+	h := DCOSHost{}
 
 	if len(ipCommand) == 0 {
-		return a, fmt.Errorf("Must pass ipCommand to NewAgent()")
+		return h, fmt.Errorf("Must pass ipCommand to DCOSHost()")
 	}
 	if port < 1024 {
```

Comment:
usually the pattern we are following is a constructor function `New...` returns an interface and `DCOSHost` is an implementation. Also all the parameters probably should be implemented as functional parameters. This will help you extend the implementation easily without changing the interface and all callers. This is also a great way validating your input parameters.